services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,49 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if(!event){
+    return ERR_CODE_INVALID_ARG;
+  }
+
 
+  if(xQueueSend(thermalMgrQueueHandle, (const void *) event, 0) != pdTRUE){
+    return ERR_CODE_INVALID_QUEUE_MSG;
```

Comment:
this would fail in the case of the queue being full or the queue having not been created only, not for an invalid queue message since the queue itself has no way to check whether or not it is a valid message